src/asynchro.rs

```diff
@@ -416,17 +408,26 @@ where
 
         // store last index for next iteration
         self.last_index = idx - self.chunk_size as f64;
-        for chan in used_channels.iter() {
-            //for w in wave_out.iter_mut() {
-            wave_out[*chan].truncate(n);
+        for (chan, active) in active_channels_mask.iter().enumerate() {
+            if *active {
+                wave_out[chan].truncate(n);
+            }
         }
+        #[cfg(debug_assertions)]
         trace!(
             "Resampling channels {:?}, {} frames in, {} frames out",
-            used_channels,
+            active_channels_mask,
             self.chunk_size,
             n,
         );
-        Ok(wave_out)
+        Ok(())
+    }
+
+    fn get_max_output_size(&self) -> (usize, usize) {
+        (
+            self.nbr_channels,
+            (self.chunk_size as f64 * self.resample_ratio + 10.0) as usize,
```

Comment:
Where does this magic number `10.0` come from?

Reply:
In principle +1 should be enough, but I added a few more to be on the safe side. The number is pretty arbitrary.

Reply:
If you're going to keep this function -- which I don't think is a good idea if max output size isn't deterministic here -- then it would help to add a comment explaining the reasoning behind picking `10.0`.